### PR TITLE
[RHELC-1293] Automatically assign pytest markers, ignore warnings

### PR DESCRIPTION
**Pull request by danmyway**
* assign pytest markers automatically
  * https://docs.pytest.org/en/latest/example/markers.html#automatically-adding-markers-based-on-test-names
* ignore PytestUnknownWarning on custom markers starting with test_
  * https://docs.python.org/3/library/warnings.html#warnings.filterwarnings
* remove unused imports
* remove custom markers from both function decorators and pytest.ini
* change some test function names 
* remove a kernel module test verifying, that unloading kernel module will not yield previously appearing inhibitor, as it has litttle to no value

Jira Issues:

- [RHELC-1293](https://issues.redhat.com/browse/RHELC-1293)

Checklist

- [x] PR has been tested manually in a VM (either author or reviewer)
- [x] Jira issue has been made public if possible
- [x] `[RHELC-]` or `[HMS-]` is part of the PR title 
- [x] GitHub label has been added to help with Release notes 
- [x] PR title explains the change from the user's point of view
- [x] Code and tests are documented properly
- [x] The commits are squashed to as few commits as possible (without losing data) 
- [x] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

File: tests/integration/tier1/destructive/firewalld-disabled-ol8/test_firewalld_disabled_ol8.py [hunk 1]
```diff
@@ -8,7 +8,6 @@
 FIREWALLD_CONFIG_FILE = "/etc/firewalld/firewalld.conf"
 
 
-@pytest.mark.test_firewalld_disabled
 def test_firewalld_disabled_ol8(shell, convert2rhel):
```
Comment (Venefilyn) [inline tests/integration/tier1/destructive/firewalld-disabled-ol8/test_firewalld_disabled_ol8.py:11]: What's the reason for removing all the markers? Are they not needed?

Comment (r0x0d) [reply tests/integration/tier1/destructive/firewalld-disabled-ol8/test_firewalld_disabled_ol8.py:11]: Yeah, as per documentation it is supposed to be done automatically by pytest via the hook they have: https://docs.pytest.org/en/latest/example/markers.html#automatically-adding-markers-based-on-test-names

Comment (danmyway) [reply tests/integration/tier1/destructive/firewalld-disabled-ol8/test_firewalld_disabled_ol8.py:11]: Yep, it is indeed, as Rodolfo says.
I'm applying the suggestion from #1018 
There is also a way to automatically register the marker for the sake of the run during the `pytest_sessionstart()` test collection, but that I see as unnecessary.
Simply ignoring the markers based on the custom prefix and registering those without one is good enough.

File: tests/integration/conftest.py [hunk 2]
```diff
@@ -29,6 +31,23 @@
 logging.basicConfig(level=os.environ.get("DEBUG", "INFO"), stream=sys.stderr)
 logger = logging.getLogger(__name__)
 
+
+def pytest_collection_modifyitems(items):
```
Comment (kokesak) [inline tests/integration/conftest.py:35]: great description, thanks!

**Comment by kokesak** [comment]
rebase and rerun please

**Review by kokesak — APPROVED** [review]
LGTM!

**Pull request merged**